modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobContainer.java

```diff
@@ -296,10 +296,20 @@ long getLargeBlobThresholdInBytes() {
         return blobStore.bufferSizeInBytes();
     }
 
+    @Override
+    public void writeBlobAtomic(
+        OperationPurpose purpose,
+        String blobName,
+        InputStream inputStream,
+        long blobSize,
+        boolean failIfAlreadyExists
+    ) throws IOException {
+        writeBlob(purpose, blobName, inputStream, blobSize, failIfAlreadyExists);
+    }
+
     @Override
     public void writeBlobAtomic(OperationPurpose purpose, String blobName, BytesReference bytes, boolean failIfAlreadyExists)
         throws IOException {
-        assert BlobContainer.assertPurposeConsistency(purpose, blobName);
```

Comment:
Is it intentional to remove this assertion? It seems to me that regardless of which `writeBlobAtomic` method we use, `OperationPurpose` and blob name stay the same and can be checked for consistency?

Reply:
Yes it's intentionally removed because I found it superfluous because the `writeBlob` function it calls already does `assert assertPurposeConsistency(purpose, blobName);` in its beginning.